samples/dnn/gpt2_inference.py

```diff
@@ -0,0 +1,108 @@
+'''
+This is a sample script to run GPT-2 inference in OpenCV using ONNX model.
+The script loads the GPT-2 model and runs inference on a given prompt.
+Currently script only works with fixed size window, that means
+you will have to specify prompt of the same length as when model was exported to ONNX.
+
+
+Exporting GPT-2 model to ONNX.
+To export GPT-2 model to ONNX, you can use the following procedure:
+
+1. Clone fork of Andrej Karpathy's GPT-2 repository:
+
+    git clone https://github.com/Abdurrahheem/build-nanogpt/tree/ash/export-gpt2-onnx
+
+2. Install the required dependencies:
+
+    pip install -r requirements.txt
+
+3  Export the model to ONNX:
+
+    python export2onnx.py --promt=<Any-promt-you-want> --batch_size=<batch-size>
+
+
+Run the script:
+
+    python gpt2_inference.py --model=<path-to-onnx-model> --max_seq_len=<max-output-lenght> --batch_size=<use-one-used-while-exportinh> --prompt=<use-promt-of-the-same-length-used-while-exporting>
```

Comment:
> --prompt=<use-promt-of-the-same-length-used-while-exporting>

Can sample support simple padding in case of shorter prompt? Otherwise user cannot understand how to fill the same length.

Reply:
Currenly no

Reply:
how to know head of time which prompt legth user used?

Reply:
If model exported with some input shape we can accept prompt with less or equal length and pad in case of shorter prompt. There is an option to obtain input shape in OpenCV like this: https://github.com/opencv/opencv/blob/6a11847d579e95bc5cca64d951684d4b578870ae/modules/dnn/src/model.cpp#L51

Reply:
Let's use `max_input_length` as an additional parameter: https://github.com/search?q=repo%3Ahuggingface%2Ftransformers%20max_input_length&type=code